In [1]:
import numpy as np
import sympy as sym
import matplotlib.pyplot as plt
import seaborn as sns
sns.set_style("ticks")
import pandas as pd
%matplotlib inline

El costo total (en miles de pesos) de pedido y almacenaje de x automoviles es de 

$C(x) = 4x + 720 + 921600/x $

Determine el tamaño del pedido que minimiza el costo total 

In [2]:
sym.var('x', real = True)

x

In [3]:
def c(x):
    return 4*x + 720 + 921600/x

Sacamos la primer derivada 

In [4]:
dc = sym.diff(c(x) ,x)
dc

4 - 921600/x**2

Después sacamos los puntos críticos igualando a cero nuestra primer derivada 

In [10]:
IgualamosACero = sym.solve(dc, x)
IgualamosACero

[-480, 480]

Como no puede existir un pedido de carros negativos, tomamos el número positivo

Donde $x = 480$ es el número de carros que debe tener el pedido para minimizar el costo 

In [18]:
print ("$C(480)$" , 4*480 + 720 + 921600/480)

$C(480)$ 4560.0


Donde 4560 es el costo mínimo para producir el pedido 